dotnet/src/Agents/OpenAI/Internal/AssistantToolResourcesFactory.cs

```diff
@@ -26,24 +26,23 @@ internal static class AssistantToolResourcesFactory
 
         if (hasVectorStore || hasCodeInterpreterFiles)
         {
-            toolResources =
-                new ToolResources()
+            var fileSearch = hasVectorStore
+                ? new FileSearchToolResources
                 {
-                    FileSearch =
-                        hasVectorStore ?
-                            new FileSearchToolResources()
-                            {
-                                VectorStoreIds = [vectorStoreId!],
-                            } :
-                            null,
-                    CodeInterpreter =
-                        hasCodeInterpreterFiles ?
-                            new CodeInterpreterToolResources()
-                            {
-                                FileIds = (IList<string>)codeInterpreterFileIds!,
-                            } :
-                            null,
-                };
+                    VectorStoreIds = { vectorStoreId! }
+                }
+                : null;
+
+            var codeInterpreter = hasCodeInterpreterFiles
+                ? new CodeInterpreterToolResources()
+                : null;
+
+            codeInterpreter?.FileIds.AddRange(codeInterpreterFileIds!);
+            toolResources = new ToolResources
+            {
+                FileSearch = fileSearch,
+                CodeInterpreter = codeInterpreter
+            };
```

Comment:
I see why/how this changed.  Thank you.  Would you mind using the following convention for consistency?

```suggestion
            FileSearchToolResources? fileSearch = hasVectorStore
                ? new()
                {
                    VectorStoreIds = { vectorStoreId! }
                }
                : null;

            CodeInterpreterToolResources? codeInterpreter = hasCodeInterpreterFiles
                ? new()
                : null;
            codeInterpreter?.FileIds.AddRange(codeInterpreterFileIds!);

            toolResources = new ToolResources
            {
                FileSearch = fileSearch,
                CodeInterpreter = codeInterpreter
            };
```